plugins/obs-webrtc/webrtc/src/output.rs

```diff
@@ -0,0 +1,378 @@
+        let gather_ips = if_addrs::get_if_addrs()
+            .unwrap_or_else(|_| Vec::new())
+            .iter()
+            .fold(Vec::new(), |mut gather_ips, addr| {
+                if !addr.is_loopback() {
+                    trace!(
+                        "Valid gather address for interface {}: {:#?}",
+                        addr.name,
+                        addr.ip()
+                    );
+                    gather_ips.push(addr.ip());
+                } else {
+                    trace!(
+                        "Loopback address ignored for interface {}: {:#?}",
+                        addr.name,
+                        addr.ip()
+                    );
+                }
+                gather_ips
+            });
+
+        let mut setting_engine = SettingEngine::default();
+
+        // Only attempt to limit if we found valid interfaces, otherwise do not attempt
+        // to limit as we may be on a platform that doesn't expose enough information
+        if !gather_ips.is_empty() {
+            debug!("Gather addresses: {gather_ips:#?}");
+            setting_engine.set_ip_filter(Box::new(move |ip: std::net::IpAddr| -> bool {
+                gather_ips.contains(&ip)
+            }));
+        }
+
+        let api = APIBuilder::new()
+            .with_media_engine(m)
+            .with_interceptor_registry(registry)
+            .with_setting_engine(setting_engine)
+            .build();
+
+        // Prepare the configuration
+        let config = RTCConfiguration {
+            ..Default::default()
+        };
+
+        let peer_connection = Arc::new(api.new_peer_connection(config).await?);
+        let stats: Arc<RwLock<OutputStreamStats>> = Arc::new(RwLock::new(OutputStreamStats {
+            connect_time: None,
+            ..Default::default()
+        }));
+
+        let (worker_tx, worker_rx): (UnboundedSender<Message>, UnboundedReceiver<Message>) =
+            mpsc::unbounded_channel();
+        let error_callback: Arc<Mutex<Option<ErrorCallback>>> = Arc::new(Mutex::new(None));
+
+        let mut output_stream = Self {
+            audio_track,
+            video_track,
+            peer_connection,
+            stats,
+            worker_tx,
+            worker_handle: Arc::new(Mutex::new(None)),
+            whip_resource: Arc::new(Mutex::new(None)),
+            error_callback,
+        };
+
+        output_stream.start_worker(worker_rx);
+
+        Ok(output_stream)
+    }
+
+    pub async fn connect(&self, url: &str, bearer_token: Option<&str>) {
+        self.connect_internal(url, bearer_token)
+            .await
+            .unwrap_or_else(|e| {
+                error!("Failed connecting to: {e}");
+                let _ = self
+                    .worker_tx
+                    .send(Message::Error(OutputStreamError::ConnectFailed));
+            })
+    }
+
+    async fn connect_internal(&self, url: &str, bearer_token: Option<&str>) -> Result<()> {
+        println!("Setting up webrtc!");
+
```

Comment:
I think using `use webrtc::rtp_transceiver::rtp_transceiver_direction::RTCRtpTransceiverDirection` (inside of this function; and then using `RTCRtpTransceiverDirection` directly) would make the code in this function more readable.

Reply:
Agreed, not sure how that got in

Reply:
Fixed in 68e7a750979901ecc7ba323876c692390dac0756